### PR TITLE
Cleanup output for easy import into visualization tools

### PR DESCRIPTION
**Pull request by reidmv**
This commit rewrites the metrics querying scripts to format the output in a way that is easy to import into visualization tools like Graphite/Grafana. By necessity it switches the scripts from bash to ruby, so that hashes can be built and manipulated.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -27,4 +27,10 @@
     hosts         => $puppetdb_hosts,
   }
 
+  # Ensure remanants of cron jobs and files from older versions of this module
+  # are cleaned up.
+  pe_metric_curl_cron_jobs::pe_metric { 'puppet_server' :
+    ensure => 'absent',
```
Comment (npwalker) [inline manifests/init.pp]: Should be `metric_ensure`

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Pe_metric_curl_cron_jobs::Pe_metric[puppet_server]: has no parameter named 'ensure' at /etc/puppetlabs/code/environments/production/modules/pe_metric_curl_cron_jobs/manifests/init.pp:32 on node master201643-centos.puppetdebug.vlan
```

**Comment by djjudas21** [comment]
Works for me :+1:

**Comment by reidmv** [comment]
@npwalker any update?

**Comment by npwalker** [comment]
@reidmv I've asked Charlie to take a look at it then I think we can merge.

**Comment by Sharpie** [comment]
👍 looks fine to me!

**Comment by djjudas21** [comment]
The cleanup script which runs nightly runs find `'/opt/puppetlabs/pe_metric_curl_cron_jobs' -type f -mtime 3 -delete -print` to delete old reports. The `-print` flag is causing the `find` command to produce output like the following on stdout - which is then emailed to us:

```
/opt/puppetlabs/pe_metric_curl_cron_jobs/puppetserver/127.0.0.1/20170304T025501Z.json
/opt/puppetlabs/pe_metric_curl_cron_jobs/puppetdb/127.0.0.1/20170304T025501Z.json
```

We don't really care what it deleted - so can we remove `-print` from that cronjob and save a few emails per night? Happy to send a PR myself but it might be easier to tack another commit onto this one.

**Pull request merged**